Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -330,6 +490,150 @@ lemma iteratedFDeriv_fourierIntegral {N : ℕ∞}
   ext w : 1
   exact ((hasFTaylorSeriesUpTo_fourierIntegral L hf h'f).eq_iteratedFDeriv hn w).symm
 
+/-- The Fourier integral of the `n`-th derivative of a function is obtained by multiplying the
+Fourier integral of the original function by `(2πI L w ⬝ )^n`. -/
+theorem fourierIntegral_iteratedFDeriv [FiniteDimensional ℝ V]
+    {μ : Measure V} [Measure.IsAddHaarMeasure μ] {N : ℕ∞} (hf : ContDiff ℝ N f)
+    (h'f : ∀ (n : ℕ), n ≤ N → Integrable (iteratedFDeriv ℝ n f) μ) {n : ℕ} (hn : n ≤ N) :
+    fourierIntegral 𝐞 μ L.toLinearMap₂ (iteratedFDeriv ℝ n f)
+      = (fun w ↦ fourierPowSMulRight (-L.flip) (fourierIntegral 𝐞 μ L.toLinearMap₂ f) w n) := by
+  induction n with
+  | zero =>
+    ext w m
+    simp only [iteratedFDeriv_zero_apply, Nat.zero_eq, fourierPowSMulRight_apply, pow_zero,
+      Finset.univ_eq_empty, ContinuousLinearMap.neg_apply, ContinuousLinearMap.flip_apply,
+      Finset.prod_empty, one_smul, fourierIntegral_continuousMultilinearMap_apply' ((h'f 0 bot_le))]
+  | succ n ih =>
+    ext w m
+    -- instance on next line should not be necessary, but proof breaks down without it.
+    let NS : NormedSpace ℝ (V [×n]→L[ℝ] E) := by infer_instance
+    have J : Integrable (fderiv ℝ (iteratedFDeriv ℝ n f)) μ := by
+      specialize h'f (n + 1) hn
+      simp_rw [iteratedFDeriv_succ_eq_comp_left] at h'f
+      exact (LinearIsometryEquiv.integrable_comp_iff _).1 h'f
+    suffices H : (fourierIntegral 𝐞 μ L.toLinearMap₂ (fderiv ℝ (iteratedFDeriv ℝ n f)) w)
+          (m 0) (Fin.tail m) =
+        (-(2 * π * I)) ^ (n + 1) • (∏ x : Fin (n + 1), -L (m x) w) • ∫ v, 𝐞 (-L v w) • f v ∂μ by
+      rw [fourierIntegral_continuousMultilinearMap_apply' (h'f _ hn)]
+      simp only [iteratedFDeriv_succ_apply_left, fourierPowSMulRight_apply,
+        ContinuousLinearMap.neg_apply, ContinuousLinearMap.flip_apply]
+      rw [← fourierIntegral_continuousMultilinearMap_apply' ((J.apply_continuousLinearMap _)),
+          ← fourierIntegral_continuousLinearMap_apply' J]
+      exact H
+    have h'n : n < N := lt_of_lt_of_le (by simp [-Nat.cast_add]) hn
+    rw [fourierIntegral_fderiv]
+    · have A : ∀ (x : ℝ) (v : E), x • v = (x : ℂ) • v := fun x v ↦ rfl
+      simp only [ih h'n.le, fourierSMulRight_apply, ContinuousLinearMap.neg_apply,
+        ContinuousLinearMap.flip_apply, neg_smul, smul_neg, neg_neg, smul_apply,
+        fourierPowSMulRight_apply, A, smul_smul]
```

Comment:
there's already a lemma for `A`, although it needs a tiny bit of help to get the types right:
```suggestion
    · simp only [ih h'n.le, fourierSMulRight_apply, ContinuousLinearMap.neg_apply,
        ContinuousLinearMap.flip_apply, neg_smul, smul_neg, neg_neg, smul_apply,
        fourierPowSMulRight_apply, ← coe_smul (E := E), smul_smul]
```